repair/repair_task.hh

```diff
@@ -62,7 +62,34 @@ protected:
 };
 
 class repair_module : public task_manager::module {
+    // TODO Deexie:
+    // _status - the difference between _status and modules maps that makes us keep this map is that the information about tasks that failed
+    // is kept forever (successful one are deleted after they are finished). Thus, we can keep the map as it is or keep a set of sequence numbers of failed
+    // tasks. The latter would make as keep another map<sequence_number, task_ptr> to easily find the task by its sequence_number.
+    //
+    // _repairs - lw_shared_ptr<repair_info> can be probably replaced with shard repair task ptrs. In this case problem with shards arises, since each
+    // repair_info is kept on shard 0. Then if _repairs is changed in such a way, we would need to also replaces loops with map_reduce etc.
+    //
+    // Note that there is an option to abort repair outside of the task_manager api, where it is unabortable. It should probably be marked somehow
+    // not to cause the confusion.
+    //
+    // Where should root task start? We have two sensible possibilities:
+    // - where repair_service::do_repair_start starts (and respectively for repair_service::do_sync_data_using_repair). This will make us record all the tasks 
+    //  even the ones that were terminated in preliminary correctness check. This way we will also avoid inconvenience caused by the task id being logged before
+    //  the task is actually created.
+    // - where tracer::run starts. This way _status, _pending_repairs, and _aborted_pending_repairs could freely cooperate with task_manager::module::task.
+    // The problem with logs order appears (some data is logged in the background).
+    //
+    // The order of shutting down. Can we use common gate? Can we get rid of shutdown methods?
```

Comment:
I think we should use a the module gate, along with integrating with the task manager abort source.